### ISSUE TITLE
warning: optimization flag '-fno-strength-reduce' is not supported [-Wignored-optimization-argument]

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1225]
```
$ ARCH=m68k CROSS_COMPILE=m68k-linux-gnu- make CC=clang -j71 defconfig
$ ARCH=m68k CROSS_COMPILE=m68k-linux-gnu- make CC=clang -j71
...
clang-12: warning: optimization flag '-fno-strength-reduce' is not supported [-Wignored-optimization-argument]
```
This is an easy fix/good first issue. The kernel builds with some GCC flag to disable some GCC specific optimization (likely something is broken in that pass, or that pass exposes some brittle code).  This flag should be wrapped in the commonly used `cc-option` macro used in Kbuild. cc @mshockwave.

**Comment by mshockwave** [comment]
Managed to fix this and #1224 pretty easily, but since this will be a linux kernel change, should I just open a new branch (probably `m68k`) in this repository to host my changes?

**Comment by nathanchance** [comment]
Sure. That will make it easy for us to test and review before they get sent to the mailing list.

**Comment by nickdesaulniers** [comment]
Reopening until this is fixed in the upstream linux kernel proper.

**Comment by nathanchance** [comment]
Someone has proposed deleting this flag because it actually does nothing (see the GCC commit in the commit message): https://lore.kernel.org/r/baa95d7235921dff23bed6320518f3fa90396603.1612681353.git.fthain@telegraphics.com.au/

**Comment by mshockwave** [comment]
> Someone has proposed deleting this flag because it actually does nothing (see the GCC commit in the commit message): https://lore.kernel.org/r/baa95d7235921dff23bed6320518f3fa90396603.1612681353.git.fthain@telegraphics.com.au/

Good to know, thanks!

**Comment by nathanchance** [comment]
Patch was picked up for 5.14: https://git.kernel.org/geert/linux-m68k/c/bc7c9993a0d836aa88aca2969dcb9b22031924d3

**Comment by nickdesaulniers** [comment]
fixed: bc7c9993a0d836aa88aca2969dcb9b22031924d3
